### DISCUSSION TITLE
[Bug]: Configuration Editor sometimes has duplicate tooltips

### DISCUSSION THREAD

**Discussion by xconverge** [blakeblackshear/frigate#13836] [Report a Bug] [answered]
### Checklist

- [X] I have updated to the latest available Frigate version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.
- [X] I have tried reproducing the issue in [incognito mode](https://www.computerworld.com/article/1719851/how-to-go-incognito-in-chrome-firefox-safari-and-edge.html) to rule out problems with any third party extensions or plugins I have installed.

### Describe the problem you are having

Duplicate hover tooltips in the Configuration Editor page

### Steps to reproduce

1. Navigate to configuration editor with the settings cog icon navigation
1. Hover a configuration setting in the yaml file, notice there is 1 tooltip
1. Click "Live" to navigate to live view page
1. Navigate to configuration editor with the settings cog icon navigation
1. Hover a configuration setting in the yaml file, notice there are 2 tooltips
1. Click "Live" to navigate to live view page
1. repeat and it grows

Refreshing the configuration page explicitly can fix it

### Version

0.15.0-27e71eb

### In which browser(s) are you experiencing the issue with?

Firefox and Chrome

### Frigate config file

```yaml
Unnecessary
```

### docker-compose file or Docker CLI command

```yaml
Unnecessary
```

### Relevant Frigate log output

```shell
Unnecessary
```

### Relevant go2rtc log output

```shell
Unnecessary
```

### Operating system

Debian

### Install method

Docker Compose

### Network connection

Wired

### Camera make and model

Unnecessary

### Screenshots of the Frigate UI's System metrics pages

Unnecessary

**Comment by hawkeye217** [comment]
Can you show a screenshot or screengrab? I don't see tooltips in the editor other than over links in Firefox or Chrome.

**Reply by xconverge** [reply]
here is me doing the steps 3 times, refreshing brings it back to a single suggestion/documentation tooltip

![image](https://github.com/user-attachments/assets/47e2c7e3-b5f8-4d21-b16d-6155a511eee6)

**Reply by xconverge** [reply]
Some are more useful than others, I noticed the feature existed first with "hwaccel_args" for ffmpeg. Some of them also have a weird ||

In general the tooltips aren't really helpful currently (vs the documentation pages and reference configs), but that would be a separate issue entirely, this just screamed react to me and should be reproducible/easily actionable (albeit super low priority)

![image](https://github.com/user-attachments/assets/42dab666-2ff7-40fc-ba53-38245a3c8c10)

**Reply by hawkeye217** [reply]
Thanks. I had a `# yaml-language-server: ...` directive at the top of my config and wasn't getting tooltips. But I removed it and now I see tooltips appearing.

However, whenever I move to the sidebar to select the Live view, the tooltip disappears.

**Reply by xconverge** [reply]
yea, then navigate back to the config page, and there should be 2x tooltips now (as long as you dont do a full refresh in the browser)

**Reply by NickM-27** [reply]
the tooltips are automatically controlled and shown with data from the schema by the vscode editor so this is likely something that is a bug in there

**Reply by hawkeye217** [reply]
I can't seem to reproduce. But if you're seeing it, this may be a bug in the [monaco-editor](https://github.com/microsoft/monaco-editor) package we use for the config editor.

**Reply by xconverge** [reply]
Can one of you check if this useEffect is firing and parsing again or doing something maybe? (Maybe configureMonacoYaml is registering a hover provider each time)

https://github.com/blakeblackshear/frigate/blob/dev/web/src/pages/ConfigEditor.tsx#L88

I am fine if you just close this and ignore it, but I can very easily repro it across multiple browsers

I didn't even consider/look at the code/dependencies before opening this

**Reply by hawkeye217** [reply]
Sounds like you are halfway there to fixing the bug :)

I still can't reproduce it, unfortunately. Bizarre.

**Comment by hawkeye217** [comment, marked as answer]
Fixed in https://github.com/blakeblackshear/frigate/pull/13841

**Reply by xconverge** [reply]
I am in PST so didn't even get a chance at it yet! Nice job
